stress_test/controller.py

```diff
@@ -0,0 +1,402 @@
+        self.of_port = test_config['controller_port']
+        self.cpu_shares = test_config['controller_cpu_shares']
+        self.logs_dir = self.base_dir + test_config['controller_logs_dir']
+
+        self.build_hnd = self.base_dir + test_config['controller_build_handler']
+        self.start_hnd = self.base_dir + test_config['controller_start_handler']
+        self.stop_hnd = self.base_dir + test_config['controller_stop_handler']
+        self.status_hnd = self.base_dir + test_config['controller_status_handler']
+        self.clean_hnd = self.base_dir + test_config['controller_clean_handler']
+
+        self.status = 'UNKNOWN'
+
+        self.java_opts = ' '.join(test_config['java_opts'])
+        self.pid = -1
+
+        self._init_ssh = None
+
+        self_node = conf_collections_util.node_parameters(self.name,
+                                      self.ip,
+                                      int(self.ssh_port),
+                                      self.username,
+                                      self.password)
+
+    def init_ssh(self)
+        logging.info(
+            '[open_ssh_connection] Initiating SSH session with {0} node.'.
+            format(self.name))
+        self._init_ssh = util.netutil.ssh_connect_or_return(connection, 10)
+
+
+
+    def cleanup(self):
+        """Wrapper to the controller cleanup handler
+        """
+        logging.info('[Controller] Cleaning up')
+
+        self.status = 'CLEANING'
+        common.command_exec_wrapper([self.clean_hnd],
+                                    '[controller.clean_handler]', self._init_ssh)
+        self.status = 'CLEANED'
+
+    def check_status(self):
+        """Wrapper to the controller status handler
+        """
+        logging.info('[Controller] Checking the status')
+
+        common.command_exec_wrapper([self.status_hnd],
+
+
+    def check_other_controller(self):
+        """Checks for processes listening on the specified port
+
+        :raises Exception: when another process listens on controller's port.
+        """
+
+        logging.info(
+            '[Controller] Checking if another process is '
+            'listening on specified port. Port number: {0}.'.
+            format(self.of_port))
+
+        # check if any process listens on controller port
+        gpid = util.process.getpid_listeningonport(self.of_port, self._init_ssh)
+
+        if gpid != -1:
+            raise Exception('[check_other_controller] Another process is '
+                            'active on port {0}'.
+                            format(self.of_port))
+
+    def restart(self):
+        """Restarts the controller
+        :rtype: int
+        """
+        logging.info('[Controller] Restarting')
+
+        self.status = 'RESTARTING'
+        self.stop()
+        self.status = 'STOPPED'
+        self.pid = self.start(controller_cpus)
+        self.status = 'STARTED'
+
+    def start(self, controller_cpus):
```

Comment:
We will take a big decision here and __remove the use of `controller_cpus`__. 

Reasons: 
- it's currently buggy (it computes the CPUs to run the controller based not on the controller host, but on the NSTAT node)
- it would be better to apply CPU binding for the controller not internally, but externally. For example, if we run the controller within a VM, it would be best to bind the VMs virtual CPUs on the physical machine's CPUs, and not bind the controller's threads to the virtual CPUs.